tests/integration/components/conversations/conversation-part-closed-test.js

```diff
@@ -27,18 +27,42 @@ moduleForComponent('conversations/conversation-part-closed', 'Integration | Comp
   }
 });
 
-test('if current user closes message, "You closed this at" is rendered', function(assert) {
+test('if current user closes message, "You closed this" is rendered', function(assert) {
   assert.expect(1);
 
   let user = {
-    id: 1
+    id: 1,
+    username: 'testuser'
   };
 
+  let closedAt = 'MM/DD/YYYY hh:mm:ss';
+
   set(this, 'author', user);
-  set(this, 'closedAt', moment().subtract(2, 'days'));
+  set(this, 'closedAt', moment().fromNow());
   stubService(this, 'current-user', { user: { id: 1 } });
 
   renderPage();
+  assert.equal(page.closedAt.text, 'You closed this on', 'The closed at timestamp is rendered');
+
+});
+
+test('if someone other than the current user closes the message, "Author.username closed this at" is rendered', function(assert) {
+  assert.expect(1);
+
+  let user = {
+    id: 1,
+    username: 'currentuser'
+  };
+
+  let user1 = {
+    id: 2,
+    username: 'authoruser'
+  };
+
+  set(this, 'author', { user1 });
+  set(this, 'closedAt', moment().from());
+  stubService(this, 'current-user', { user });
```

Comment:
I also can't get the "author" nor the "time" to pass in here, either. Been trying for too long so moving onto something else.

Reply:
There's a misunderstanding here with the author.

`{ user }` is an object shorthand. Basically, it's a shorter way of writing `{ user: user }`

With that in mind

```Javascript
set(this, 'author', { user1 });
```

means you're actually doing

```Javascript
set(this, 'author', { user1: user1 });
```

which clearly wont work. 

In your setup, there's no `author.username`, `author.id`, etc., there's 'author.user1.username`, `author.user1.id`, etc.

Thus, what you need to do instead is, 

```Javascript
set(this, 'author', user1);
```

Meaning, the value of `author` in our component is the same as `user1`

What might be confusing you is the 

```Javascript
stubService(this, 'current-user', { user });
```

but we actually do a different thing there.

If you take a look at how the actual `current-user` service is implemented, under `app/services/current-user.js`, you will see that it's an `Ember.Service`, with a `user` property in it, so what we're doing is exactly what we want to do - replace the actual service with a fake one, where the value of `user` is our user.

That means, we will have `currentUser.user`, `currentUser.user.id`, `currentUser.user.username`, etc., exactly as we want.

Reply:
As for the time stamp, look for an example in  `admin-github-event-test.js`

The general idea is to get some timestamp, either using `Date.now()` or `moment()`. Both will return the exact current time. The example above uses `moment()`.

Then, store that time stamp into a variable. This is **critical**. You can't just go and do the following, for example:

```Javscript
let timestamp = moment();
assert.equal(timestamp, moment());
```

The simple reason you can't do this is that both `Date.now()` and `moment()` return a different value every time you call them - the **current** timestamp, meaning the exact time at the time the function was called.

So if we want to test our code here, we need to understand what our component is supposed to do - The intent of the test is to ensure that the component does what it supposed to do.

Our component is supposed to tell us who closed our conversation and when.

The who will be either the username of the author, or just "You" if the author is the current user, so we need two tests for that, which I think you'll know how to cover.

The "when" will be the time of the closing event, meaning the time the conversation part of "`partType`" `"closed"` was created (or, in terms of the code, the value of `closedAt`). We also want the `when` to nicely formatted in some manner, which is something I described in my comment above.

To give a few examples for the username part

- If the current user is `begedin`, and the person who closed the conversation is `daveconnis`, we want the text to contain `"daveconnis closed this"`
- If the current user is `daveconnis` and `daveconnis` closed the conversation, we want the text to say `"You closed this"`

As for the timestamp, it gets a bit more complicated because you sort of need to know how the friendly formatting works, but what you would do is something like

- Create a couple of timestamps at varying moments in the past
- For each of them
  - set the value of `closedAt` to that
  - check if the text contains the friendly version of that timestamp
  - due to some weird ember internals, you need to use the run loop to do that, otherwise it won't update

As a concrete example

```Javascript
// moment() returns the current time. `subtract(amount, type)` then modifies that
let twoMinutesAgo = moment().subtract(2, 'minutes');
// from() is the friendly formatted version of the above, same as what the {{moment-from-now}} helper does
let twoMinutesAgoFriendly = twoMinutesAgo.from();
// this is where we use the run loop to set the new value
run(() => { set(this, 'closedAt', twoMinutesAgo) }
// indexOf will return -1 if the text does not contain twoMinutesAgoFriendly
// If it does, it will contain the position in the text where it starts, so 0 or higher
// checking if it's higher than -1 means we check if the text contains it
assert.ok(page.closedAt.text.indexOf(twoMinutesAgoFriendly) > -1) 
```
Then you do the same thing with, for example, two seconds ago, 2 days ago, etc. Actually, just those 3 examples ought to be enough in total.